docs/source/developers/guide/step_by_step/testing.rst

```diff
@@ -62,29 +62,101 @@ In this section we outline steps needed for unit testing in Arrow.
 
       If the tests start failing, try to recompile
       PyArrow or C++.
-      
+
       .. note::
 
          **Recompiling Cython**
 
          If you only make changes to `.py` files, you do not need to
          recompile PyArrow. However, you should recompile it if you make
          changes in `.pyx` or `.pxd` files.
-        
+
          To do that run this command again:
 
          .. code:: console
 
             $ python setup.py build_ext --inplace
 
       .. note::
-		
+
          **Recompiling C++**
 
          Similarly, you will need to recompile the C++ code if you have
          made changes to any C++ files. In this case,
-         re-run the cmake commands again. 
+         re-run the cmake commands again.
 
    .. tab:: R tests
 
-      .. TODO
+      We use `testthat <https://testthat.r-lib.org/index.html>`_ for unit testing in R. More specifically, we use the `3rd edition of testthat <https://testthat.r-lib.org/articles/third-edition.html>`_. On rare occasions we might want the behaviour of the 2nd edition of testthat, which is indicated by ``testthat::local_edition(2)``.
+
+      **Structure**
+
+      Expect the usual testthat folder structure:
+
+      .. code-block:: R
+
+         tests
+          ├── testthat      # test files live here
+          └── testthat.R    # runs tests when R CMD check runs (e.g. with devtools::check())
+
+      This is the fundamental structure of testing in R with ``testthat``. Files such as ``testthat.R`` are not expected to change very often. For the ``arrow`` R package ``testthat.R`` also defines how the results of the various tests are displayed / reported in the console.
+
+      Usually, most files in the ``R/`` sub-folder have a corresponding test file in ``tests/testthat``.
+
+      **Running tests**
+
+      To run all tests in a package locally call
+
+      .. code-block:: R
+
+         devtools::test()
+
+      in the R console. Alternatively, you can use
+
+      .. code:: console
+
+         $ make test
+
+      in the shell.
+
+      You can run the tests in a single test file you have open with
+
+      .. code-block:: R
+
+         devtools::test_active_file()
+
+      All tests are also run as part of our continuous integration (CI) pipelines.
+
+      The Arrow R Developer guide also has a section on running tests. You can check it out `here <https://arrow.apache.org/docs/r/articles/developing.html#running-tests>`_.
```

Comment:
```suggestion
      The `Arrow R Developer guide also has a section<https://arrow.apache.org/docs/r/articles/developing.html#running-tests>`_ on running tests.
```

Avoids the click here type of pattern (and is a bit more concise.

Reply:
I think I got the RST correct here, but not sure about that `_` at the end, that seems out of place, but I'm no RST expert

Reply:
I think the `_` is needed after `<>`